# This file is part of HiGFXback

# requires
REQUIRES="make-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
sh ./Configure --prefix

# build
make -f Makefile-libbz2_so

# install
install -d $DESTDIR/bin
install bzdiff $DESTDIR/bin
ln -sf bzdiff $DESTDIR/bin/bzcmp
install bzgrep $DESTDIR/bin
ln -sf bzgrep $DESTDIR/bin/bzegrep
ln -sf bzgrep $DESTDIR/bin/bzfgrep
install bzip2-shared $DESTDIR/bin/bzip2
ln -sf bzip2 $DESTDIR/bin/bunzip2
ln -sf bzip2 $DESTDIR/bin/bzcat
install bzip2recover-shared $DESTDIR/bin/bzip2recover
install bzmore $DESTDIR/bin
ln -sf bzmore $DESTDIR/bin/bzless
install -d $DESTDIR/include
install -m 644 bzlib.h $DESTDIR/include
install -d $DESTDIR/lib
install libbz2.so.1.* $DESTDIR/lib/libbz2.so.1
ln -sf libbz2.so.1 $DESTDIR/lib/libbz2.so
install -d $DESTDIR/lib/pkgconfig
install -m 644 bzip2.pc $DESTDIR/lib/pkgconfig

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/bzip2-build.pc << EOF
Name: bzip2
Version: 1.0.8
Description: bzip2 compression library
Requires: $REQUIRES

devel=\\
/include/bzlib.h \\
/lib/libbz2.so \\
/lib/pkgconfig/bzip2.pc

exec=\\
/bin/bunzip2 \\
/bin/bzcat \\
/bin/bzcmp \\
/bin/bzdiff \\
/bin/bzegrep \\
/bin/bzfgrep \\
/bin/bzgrep \\
/bin/bzip2 \\
/bin/bzip2recover \\
/bin/bzless \\
/bin/bzmore \\
/lib/libbz2.so.1
EOF

exit
--- bzip2-1.0.8.orig/Configure
+++ bzip2-1.0.8/Configure
@@ -0,0 +1,40 @@
+#!/bin/sh
+
+# ------------------------------------------------------------------
+# This file is part of bzip2/libbzip2, a program and library for
+# lossless, block-sorting data compression.
+#
+# bzip2/libbzip2 version 1.0.8 of 13 July 2019
+# Copyright (C) 1996-2019 Julian Seward <jseward@acm.org>
+#
+# Please read the WARNING, DISCLAIMER and PATENTS sections in the 
+# README file.
+#
+# This program is released under the terms of the license contained
+# in the file LICENSE.
+# ------------------------------------------------------------------
+
+PREFIX=/usr/local
+
+while : ; do
+  case "$1" in
+    "")
+      break
+    ;;
+    --prefix)
+      PREFIX=$2
+      shift
+    ;;
+    *)
+      echo Unknown option \"$1\".
+      exit 1
+    ;;
+  esac
+  shift
+done
+
+sed -i "/^PREFIX=/s#=.*#=$PREFIX#" Makefile
+
+sed -i "/^PREFIX=/s#=.*#=$PREFIX#" Makefile-libbz2_so
+
+sed -i "/^prefix=/s#=.*#=$PREFIX#" bzip2.pc
--- bzip2-1.0.8.orig/Makefile
+++ bzip2-1.0.8/Makefile
@@ -26,7 +26,6 @@
 # Where you want it installed when you do 'make install'
 PREFIX=/usr/local
 
-
 OBJS= blocksort.o  \
       huffman.o    \
       crctable.o   \
@@ -35,7 +34,7 @@
       decompress.o \
       bzlib.o
 
-all: libbz2.a bzip2 bzip2recover test
+all: libbz2.a bzip2 bzip2recover
 
 bzip2: libbz2.a bzip2.o
 	$(CC) $(CFLAGS) $(LDFLAGS) -o bzip2 bzip2.o -L. -lbz2
@@ -70,43 +69,43 @@
 	@cat words3
 
 install: bzip2 bzip2recover
-	if ( test ! -d $(PREFIX)/bin ) ; then mkdir -p $(PREFIX)/bin ; fi
-	if ( test ! -d $(PREFIX)/lib ) ; then mkdir -p $(PREFIX)/lib ; fi
-	if ( test ! -d $(PREFIX)/man ) ; then mkdir -p $(PREFIX)/man ; fi
-	if ( test ! -d $(PREFIX)/man/man1 ) ; then mkdir -p $(PREFIX)/man/man1 ; fi
-	if ( test ! -d $(PREFIX)/include ) ; then mkdir -p $(PREFIX)/include ; fi
-	cp -f bzip2 $(PREFIX)/bin/bzip2
-	cp -f bzip2 $(PREFIX)/bin/bunzip2
-	cp -f bzip2 $(PREFIX)/bin/bzcat
-	cp -f bzip2recover $(PREFIX)/bin/bzip2recover
-	chmod a+x $(PREFIX)/bin/bzip2
-	chmod a+x $(PREFIX)/bin/bunzip2
-	chmod a+x $(PREFIX)/bin/bzcat
-	chmod a+x $(PREFIX)/bin/bzip2recover
-	cp -f bzip2.1 $(PREFIX)/man/man1
-	chmod a+r $(PREFIX)/man/man1/bzip2.1
-	cp -f bzlib.h $(PREFIX)/include
-	chmod a+r $(PREFIX)/include/bzlib.h
-	cp -f libbz2.a $(PREFIX)/lib
-	chmod a+r $(PREFIX)/lib/libbz2.a
-	cp -f bzgrep $(PREFIX)/bin/bzgrep
-	ln -s -f $(PREFIX)/bin/bzgrep $(PREFIX)/bin/bzegrep
-	ln -s -f $(PREFIX)/bin/bzgrep $(PREFIX)/bin/bzfgrep
-	chmod a+x $(PREFIX)/bin/bzgrep
-	cp -f bzmore $(PREFIX)/bin/bzmore
-	ln -s -f $(PREFIX)/bin/bzmore $(PREFIX)/bin/bzless
-	chmod a+x $(PREFIX)/bin/bzmore
-	cp -f bzdiff $(PREFIX)/bin/bzdiff
-	ln -s -f $(PREFIX)/bin/bzdiff $(PREFIX)/bin/bzcmp
-	chmod a+x $(PREFIX)/bin/bzdiff
-	cp -f bzgrep.1 bzmore.1 bzdiff.1 $(PREFIX)/man/man1
-	chmod a+r $(PREFIX)/man/man1/bzgrep.1
-	chmod a+r $(PREFIX)/man/man1/bzmore.1
-	chmod a+r $(PREFIX)/man/man1/bzdiff.1
-	echo ".so man1/bzgrep.1" > $(PREFIX)/man/man1/bzegrep.1
-	echo ".so man1/bzgrep.1" > $(PREFIX)/man/man1/bzfgrep.1
-	echo ".so man1/bzmore.1" > $(PREFIX)/man/man1/bzless.1
-	echo ".so man1/bzdiff.1" > $(PREFIX)/man/man1/bzcmp.1
+	if ( test ! -d $(DESTDIR)$(PREFIX)/bin ) ; then mkdir -p $(DESTDIR)$(PREFIX)/bin ; fi
+	if ( test ! -d $(DESTDIR)$(PREFIX)/lib ) ; then mkdir -p $(DESTDIR)$(PREFIX)/lib ; fi
+	if ( test ! -d $(DESTDIR)$(PREFIX)/lib/pkgconfig ) ; then mkdir -p $(DESTDIR)$(PREFIX)/lib/pkgconfig ; fi
+	if ( test ! -d $(DESTDIR)$(PREFIX)/share/man ) ; then mkdir -p $(DESTDIR)$(PREFIX)/share/man ; fi
+	if ( test ! -d $(DESTDIR)$(PREFIX)/share/man/man1 ) ; then mkdir -p $(DESTDIR)$(PREFIX)/share/man/man1 ; fi
+	if ( test ! -d $(DESTDIR)$(PREFIX)/include ) ; then mkdir -p $(DESTDIR)$(PREFIX)/include ; fi
+	cp -f bzip2 $(DESTDIR)$(PREFIX)/bin/bzip2
+	ln -s -f bzip2 $(DESTDIR)$(PREFIX)/bin/bunzip2
+	ln -s -f bzip2 $(DESTDIR)$(PREFIX)/bin/bzcat
+	cp -f bzip2recover $(DESTDIR)$(PREFIX)/bin/bzip2recover
+	chmod a+x $(DESTDIR)$(PREFIX)/bin/bzip2
+	chmod a+x $(DESTDIR)$(PREFIX)/bin/bzip2recover
+	cp -f bzip2.1 $(DESTDIR)$(PREFIX)/share/man/man1
+	chmod a+r $(DESTDIR)$(PREFIX)/share/man/man1/bzip2.1
+	cp -f bzlib.h $(DESTDIR)$(PREFIX)/include
+	chmod a+r $(DESTDIR)$(PREFIX)/include/bzlib.h
+	cp -f libbz2.a $(DESTDIR)$(PREFIX)/lib
+	chmod a+r $(DESTDIR)$(PREFIX)/lib/libbz2.a
+	sed s@prefix=@prefix=$(PREFIX)@ bzip2.pc > $(DESTDIR)$(PREFIX)/lib/pkgconfig/bzip2.pc
+	cp -f bzgrep $(DESTDIR)$(PREFIX)/bin/bzgrep
+	ln -s -f bzgrep $(DESTDIR)$(PREFIX)/bin/bzegrep
+	ln -s -f bzgrep $(DESTDIR)$(PREFIX)/bin/bzfgrep
+	chmod a+x $(DESTDIR)$(PREFIX)/bin/bzgrep
+	cp -f bzmore $(DESTDIR)$(PREFIX)/bin/bzmore
+	ln -s -f bzmore $(DESTDIR)$(PREFIX)/bin/bzless
+	chmod a+x $(DESTDIR)$(PREFIX)/bin/bzmore
+	cp -f bzdiff $(DESTDIR)$(PREFIX)/bin/bzdiff
+	ln -s -f bzdiff $(DESTDIR)$(PREFIX)/bin/bzcmp
+	chmod a+x $(DESTDIR)$(PREFIX)/bin/bzdiff
+	cp -f bzgrep.1 bzmore.1 bzdiff.1 $(DESTDIR)$(PREFIX)/share/man/man1
+	chmod a+r $(DESTDIR)$(PREFIX)/share/man/man1/bzgrep.1
+	chmod a+r $(DESTDIR)$(PREFIX)/share/man/man1/bzmore.1
+	chmod a+r $(DESTDIR)$(PREFIX)/share/man/man1/bzdiff.1
+	echo ".so man1/bzgrep.1" > $(DESTDIR)$(PREFIX)/share/man/man1/bzegrep.1
+	echo ".so man1/bzgrep.1" > $(DESTDIR)$(PREFIX)/share/man/man1/bzfgrep.1
+	echo ".so man1/bzmore.1" > $(DESTDIR)$(PREFIX)/share/man/man1/bzless.1
+	echo ".so man1/bzdiff.1" > $(DESTDIR)$(PREFIX)/share/man/man1/bzcmp.1
 
 clean: 
 	rm -f *.o libbz2.a bzip2 bzip2recover \
@@ -114,7 +113,6 @@
 	sample1.tst sample2.tst sample3.tst
 
 blocksort.o: blocksort.c
-	@cat words0
 	$(CC) $(CFLAGS) -c blocksort.c
 huffman.o: huffman.c
 	$(CC) $(CFLAGS) -c huffman.c
--- bzip2-1.0.8.orig/Makefile-libbz2_so
+++ bzip2-1.0.8/Makefile-libbz2_so
@@ -1,11 +1,3 @@
-
-# This Makefile builds a shared version of the library, 
-# libbz2.so.1.0.8, with soname libbz2.so.1.0,
-# at least on x86-Linux (RedHat 7.2), 
-# with gcc-2.96 20000731 (Red Hat Linux 7.1 2.96-98).  
-# Please see the README file for some important info 
-# about building the library like this.
-
 # ------------------------------------------------------------------
 # This file is part of bzip2/libbzip2, a program and library for
 # lossless, block-sorting data compression.
@@ -20,40 +12,97 @@
 # in the file LICENSE.
 # ------------------------------------------------------------------
 
-
 SHELL=/bin/sh
+
+# To assist in cross-compiling
 CC=gcc
+LDFLAGS=
+
 BIGFILES=-D_FILE_OFFSET_BITS=64
-CFLAGS=-fpic -fPIC -Wall -Winline -O2 -g $(BIGFILES)
+CFLAGS=-fPIC -Wall -Winline -O2 -g $(BIGFILES)
+
+# Where you want it installed when you do 'make install'
+PREFIX=/usr/local
 
-OBJS= blocksort.o  \
-      huffman.o    \
-      crctable.o   \
-      randtable.o  \
-      compress.o   \
-      decompress.o \
-      bzlib.o
-
-all: $(OBJS)
-	$(CC) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.8 $(OBJS)
-	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.8
-	rm -f libbz2.so.1.0
-	ln -s libbz2.so.1.0.8 libbz2.so.1.0
+OBJS= blocksort.lo  \
+      huffman.lo    \
+      crctable.lo   \
+      randtable.lo  \
+      compress.lo   \
+      decompress.lo \
+      bzlib.lo
+
+all: libbz2.so.1.0.8 bzip2-shared bzip2recover-shared
+
+bzip2-shared: libbz2.so.1.0.8 bzip2.lo
+	$(CC) $(CFLAGS) $(LDFLAGS) -o bzip2-shared bzip2.lo -L. -lbz2
+
+bzip2recover-shared: bzip2recover.lo
+	$(CC) $(CFLAGS) $(LDFLAGS) -o bzip2recover-shared bzip2recover.lo
+
+libbz2.so.1.0.8: $(OBJS)
+	$(CC) -shared -Wl,-soname -Wl,libbz2.so.1 -o libbz2.so.1.0.8 $(OBJS)
+	rm -f libbz2.so libbz2.so.1.0
+	ln -s libbz2.so.1.0.8 libbz2.so
+	ln -s libbz2.so.1.0.8 libbz2.so.1
+
+install: bzip2-shared bzip2recover-shared
+	if ( test ! -d $(DESTDIR)$(PREFIX)/bin ) ; then mkdir -p $(DESTDIR)$(PREFIX)/bin ; fi
+	if ( test ! -d $(DESTDIR)$(PREFIX)/lib ) ; then mkdir -p $(DESTDIR)$(PREFIX)/lib ; fi
+	if ( test ! -d $(DESTDIR)$(PREFIX)/lib/pkgconfig ) ; then mkdir -p $(DESTDIR)$(PREFIX)/lib/pkgconfig ; fi
+	if ( test ! -d $(DESTDIR)$(PREFIX)/share/man ) ; then mkdir -p $(DESTDIR)$(PREFIX)/share/man ; fi
+	if ( test ! -d $(DESTDIR)$(PREFIX)/share/man/man1 ) ; then mkdir -p $(DESTDIR)$(PREFIX)/share/man/man1 ; fi
+	if ( test ! -d $(DESTDIR)$(PREFIX)/include ) ; then mkdir -p $(DESTDIR)$(PREFIX)/include ; fi
+	cp -f bzip2-shared $(DESTDIR)$(PREFIX)/bin/bzip2
+	ln -s -f bzip2 $(DESTDIR)$(PREFIX)/bin/bunzip2
+	ln -s -f bzip2 $(DESTDIR)$(PREFIX)/bin/bzcat
+	cp -f bzip2recover-shared $(DESTDIR)$(PREFIX)/bin/bzip2recover
+	chmod a+x $(DESTDIR)$(PREFIX)/bin/bzip2
+	chmod a+x $(DESTDIR)$(PREFIX)/bin/bzip2recover
+	cp -f bzip2.1 $(DESTDIR)$(PREFIX)/share/man/man1
+	chmod a+r $(DESTDIR)$(PREFIX)/share/man/man1/bzip2.1
+	cp -f bzlib.h $(DESTDIR)$(PREFIX)/include
+	chmod a+r $(DESTDIR)$(PREFIX)/include/bzlib.h
+	cp -d libbz2.so* $(DESTDIR)$(PREFIX)/lib
+	chmod a+rx $(DESTDIR)$(PREFIX)/lib/libbz2.so.1.0.8
+	sed s@prefix=@prefix=$(PREFIX)@ bzip2.pc > $(DESTDIR)$(PREFIX)/lib/pkgconfig/bzip2.pc
+	cp -f bzgrep $(DESTDIR)$(PREFIX)/bin/bzgrep
+	ln -s -f bzgrep $(DESTDIR)$(PREFIX)/bin/bzegrep
+	ln -s -f bzgrep $(DESTDIR)$(PREFIX)/bin/bzfgrep
+	chmod a+x $(DESTDIR)$(PREFIX)/bin/bzgrep
+	cp -f bzmore $(DESTDIR)$(PREFIX)/bin/bzmore
+	ln -s -f bzmore $(DESTDIR)$(PREFIX)/bin/bzless
+	chmod a+x $(DESTDIR)$(PREFIX)/bin/bzmore
+	cp -f bzdiff $(DESTDIR)$(PREFIX)/bin/bzdiff
+	ln -s -f bzdiff $(DESTDIR)$(PREFIX)/bin/bzcmp
+	chmod a+x $(DESTDIR)$(PREFIX)/bin/bzdiff
+	cp -f bzgrep.1 bzmore.1 bzdiff.1 $(DESTDIR)$(PREFIX)/share/man/man1
+	chmod a+r $(DESTDIR)$(PREFIX)/share/man/man1/bzgrep.1
+	chmod a+r $(DESTDIR)$(PREFIX)/share/man/man1/bzmore.1
+	chmod a+r $(DESTDIR)$(PREFIX)/share/man/man1/bzdiff.1
+	echo ".so man1/bzgrep.1" > $(DESTDIR)$(PREFIX)/share/man/man1/bzegrep.1
+	echo ".so man1/bzgrep.1" > $(DESTDIR)$(PREFIX)/share/man/man1/bzfgrep.1
+	echo ".so man1/bzmore.1" > $(DESTDIR)$(PREFIX)/share/man/man1/bzless.1
+	echo ".so man1/bzdiff.1" > $(DESTDIR)$(PREFIX)/share/man/man1/bzcmp.1
 
 clean: 
-	rm -f $(OBJS) bzip2.o libbz2.so.1.0.8 libbz2.so.1.0 bzip2-shared
+	rm -f *.lo libbz2.so* bzip2-shared bzip2recover-shared
 
-blocksort.o: blocksort.c
-	$(CC) $(CFLAGS) -c blocksort.c
-huffman.o: huffman.c
-	$(CC) $(CFLAGS) -c huffman.c
-crctable.o: crctable.c
-	$(CC) $(CFLAGS) -c crctable.c
-randtable.o: randtable.c
-	$(CC) $(CFLAGS) -c randtable.c
-compress.o: compress.c
-	$(CC) $(CFLAGS) -c compress.c
-decompress.o: decompress.c
-	$(CC) $(CFLAGS) -c decompress.c
-bzlib.o: bzlib.c
-	$(CC) $(CFLAGS) -c bzlib.c
+blocksort.lo: blocksort.c
+	$(CC) $(CFLAGS) -c -o $@ blocksort.c
+huffman.lo: huffman.c
+	$(CC) $(CFLAGS) -c -o $@ huffman.c
+crctable.lo: crctable.c
+	$(CC) $(CFLAGS) -c -o $@ crctable.c
+randtable.lo: randtable.c
+	$(CC) $(CFLAGS) -c -o $@ randtable.c
+compress.lo: compress.c
+	$(CC) $(CFLAGS) -c -o $@ compress.c
+decompress.lo: decompress.c
+	$(CC) $(CFLAGS) -c -o $@ decompress.c
+bzlib.lo: bzlib.c
+	$(CC) $(CFLAGS) -c -o $@ bzlib.c
+bzip2.lo: bzip2.c
+	$(CC) $(CFLAGS) -c -o $@ bzip2.c
+bzip2recover.lo: bzip2recover.c
+	$(CC) $(CFLAGS) -c -o $@ bzip2recover.c
--- bzip2-1.0.8.orig/bzip2.pc
+++ bzip2-1.0.8/bzip2.pc
@@ -0,0 +1,10 @@
+prefix=/usr/local
+libdir=${prefix}/lib
+includedir=${prefix}/include
+
+Name: bzip2
+Description: bzip2 compression library
+Version: 1.0.8
+
+Libs: -L${libdir} -lbz2
+Cflags: -I${includedir}
